### PR TITLE
Upgrade to Dropwizard 0.8.0-rc2 and introduce integration tests separation

### PR DESCRIPTION
**Pull request by pchojnacki**
Dropwizard 0.7.1 uses old jersey libraries which caused me a lot headache when trying to use some existing consul libraries.
My proposal is to upgrade to version 0.8.0-rc2 since Dropwizard is 'mostly' a glue between many existing libraries and in my opinion something tagged as RC-2 is generally considered stable.

Documentation for 0.8.0-rc2 can be found at
http://dropwizard.io/0.8.0-rc2/docs/manual/testing.html

@jcellary @drsnyder @nandy-andy @nmonterroso @sqreek

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -6,4 +6,8 @@ subprojects {
     apply plugin: 'java'
     sourceCompatibility = 1.8
     version = "0.1.0"
+
+    task unitTest(type: Test) {
+        exclude '**/integration/**'
```
Comment (sqreek) [inline build.gradle]: This should be in another PR or in description

**Comment by sqreek** [comment]
Its LGTM.
Please add description about you setup integration test in this pull request too.

**Comment by nmonterroso** [comment]
LGTM :ship:

**Comment by drsnyder** [comment]
> Please add description about you setup integration test in this pull request too.

+1. Please update the README with instructions on how to run unit and integration tests. We should also propose a convention. For example, where should the `integration` be added to the package? Near the tail?

It looks like the `test` task now tests everything while `unitTest` excludes anything with an integration directory in the path.
